### PR TITLE
[CI] Flake8 upgrade

### PR DESCRIPTION
**Pull request by amogkam**
Upgrade flake8 version to 3.9.1

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_runtime_env.py [hunk 1]
```diff
@@ -559,7 +559,7 @@ def test_jobconfig_compatible_3(ray_start_cluster_head, working_dir):
     script = driver_script.format(**locals())
     proc = run_string_as_driver_nonblocking(script, env)
     sleep(5)
-    runtime_env = f"""{{  "working_dir": test_module.__path__[0] }}"""
+    runtime_env = """{{  "working_dir": test_module.__path__[0] }}"""
```
Comment (amogkam) [inline python/ray/tests/test_runtime_env.py:562]:
```suggestion
    runtime_env = f"""{{  "working_dir": test_module.__path__[0] }}"""
```

**Comment by amogkam** [comment]
@sven1977 can you take a quick pass over the rllib typing changes please

**Comment by amogkam** [comment]
Bump on this @simon-mo @sven1977

**Review by simon-mo — APPROVED** [review]
LGTM

**Comment by amogkam** [comment]
`test_actor` is a known flaky test. Going to merge this.

**Pull request merged**